addons/spreadsheet_account/models/account.py

```diff
@@ -130,3 +158,28 @@ def get_account_group(self, account_types):
         )
         mapped = dict(data)
         return [mapped.get(account_type, []) for account_type in account_types]
+
+    @api.model
+    def get_residual_amount(self, args_list):
+        results = []
+        for args in args_list:
+            company_id = args['company_id'] or self.env.company.id
+            domain = self._build_move_lines_domain(args, error_domain=False)
```

Comment:
Instead of that weird `error_domain`, why don't you provide the `account_id`s to the function?

That way you can compute it here, and do one single `_read_group` with `groupby=['account_id']`